### PR TITLE
[SYCL-MLIR] Annotate handler::parallel_for arguments

### PR DESCRIPTION
**Pull request by victor-eds**
Annotate arguments to make it easier to track them when raising.

### CODE REVIEW

**Comment by victor-eds** [comment]
This will make raising these arguments easier

File: sycl/include/sycl/handler.hpp [hunk 1]
```diff
@@ -1663,21 +1669,24 @@ class __SYCL_EXPORT handler {
   }
 
   template <typename KernelName = detail::auto_name, typename KernelType>
-  void parallel_for(range<1> NumWorkItems, _KERNELFUNCPARAM(KernelFunc)) {
+  void parallel_for(range<1> NumWorkItems ANNOTATE(range),
```
Comment (sommerlukas) [inline sycl/include/sycl/handler.hpp:1672]: Do we also need to annotate the equivalent `queue` shortcut functions or do they pass through here anyway?

Comment (victor-eds) [reply sycl/include/sycl/handler.hpp:1672]: Those go through the handler ones, yep

**Review by sommerlukas — APPROVED** [review]

**Review by jopperm — APPROVED** [review]

**Review by etiotto — APPROVED** [review]
I assume this is a temporary solution. Ultimately we should not rely on annotations to determine the semantics of a SYCL construct.

**Review by whitneywhtsang — APPROVED** [review]

**Comment by victor-eds** [comment]
> I assume this is a temporary solution. Ultimately we should not rely on annotations to determine the semantics of a SYCL construct.

Yes, all this raising is a temporary solution. I'd however argue that relying on annotations might be a better solution than relying on certain patterns in the generated LLVM code.

**Pull request merged**